src/mentorship/ml/models/common.py

```diff
@@ -0,0 +1,39 @@
+from sklearn.base import clone
+
+
+class SplitPipeline:
+    def __init__(self, base_pipeline):
+        self.base_pipeline = base_pipeline
+        self.pipelines_ = {}
```

Comment:
Let's make this class more generic (as you put it in a common location and not your task-specific one like `kaggle/storesales/`). Let's add `split_key` parameter which can take either `str` or `List[str]` values:
```suggestion
    def __init__(self, base_pipeline, split_key):
        self.split_key = split_key
        self.base_pipeline = base_pipeline
        self.pipelines_ = {}
```

Reply:
[OPTIONAL] If you want to improve the performance, you can parallelize both your `fit` and `predict` methods, for example by using [`joblib.Parallel`](https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html). See an example of how `sklearn.cross_validate` method is [parallelized](https://github.com/scikit-learn/scikit-learn/blob/36958fb24/sklearn/model_selection/_validation.py#L263-L283).